lib/bazel/commands/commands.go

```diff
@@ -87,6 +91,84 @@ func NewAffectedTargetsList(parent *AffectedTargets) *AffectedTargetsList {
+		return fmt.Errorf("can't find git repo root: %v", err)
+	}
+
+	// Create temporary worktrees in which to execute bazel commands.
+	// If the end commit is not provided, use the current git directory as the end
+	// worktree, which will include uncommitted local changes.
+	startTree, err := git.NewTempWorktree(gitRoot, c.parent.Start)
+	if err != nil {
+		return fmt.Errorf("can't generate worktree for committish %q: %v", c.parent.Start, err)
+	}
+	defer startTree.Close()
+
+	endTreePath := gitRoot
+	if c.parent.End != "" {
+		endTree, err := git.NewTempWorktree(gitRoot, c.parent.End)
+		if err != nil {
+			return fmt.Errorf("can't generate worktree for committish %q: %v", c.parent.End, err)
+		}
+		defer endTree.Close()
+		endTreePath = endTree.Root()
+	}
+
+	// Open the bazel workspaces, using a well-known output_base. Since the
+	// temporary worktrees created above will have a different path on every
+	// invocation, by default bazel will create a new cache directory for them,
+	// re-download all dependencies, etc. which is both slow and will eventually
+	// fill up the disk. Reusing an output base location will ensure that at most
+	// only two are created for the purposes of this subcommand.
+	startOutputBase, err := cacheDir("affected_targets/start")
+	if err != nil {
+		return fmt.Errorf("failed to create output_base: %v", err)
+	}
+	endOutputBase, err := cacheDir("affected_targets/end")
+	if err != nil {
+		return fmt.Errorf("failed to create output_base: %v", err)
+	}
+	startWorkspace, err := bazel.OpenWorkspace(startTree.Root(), bazel.WithOutputBase(startOutputBase))
+	if err != nil {
+		return fmt.Errorf("failed to open bazel workspace for committish %q: %v", c.parent.Start, err)
+	}
+	endWorkspace, err := bazel.OpenWorkspace(endTreePath, bazel.WithOutputBase(endOutputBase))
+	if err != nil {
+		return fmt.Errorf("failed to open bazel workspace: %v", err)
+	}
+
+	// Get all target info for both VCS time points.
+	targets, err := startWorkspace.Query("deps(//...)", bazel.WithKeepGoing(), bazel.WithUnorderedOutput())
+	if err != nil {
+		return fmt.Errorf("failed to query deps for start point: %v", err)
+	}
+	// TODO(scott): Replace with logging
+	fmt.Fprintf(os.Stderr, "Processed %d targets at start point\n", len(targets))
+
+	targets, err = endWorkspace.Query("deps(//...)", bazel.WithKeepGoing(), bazel.WithUnorderedOutput())
+	if err != nil {
+		return fmt.Errorf("failed to query deps for end point: %v", err)
+	}
+	// TODO(scott): Replace with logging
+	fmt.Fprintf(os.Stderr, "Processed %d targets at end point\n", len(targets))
+
 	return fmt.Errorf("not yet implemented")
 }
 
+func cacheDir(subDir string) (string, error) {
+	d, err := os.UserCacheDir()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(d, "enkit/bazel", subDir), nil
+}
+
+func findGitRoot() (string, error) {
+	bazelWorkspace := os.Getenv("BUILD_WORKSPACE_DIRECTORY")
+	if bazelWorkspace != "" {
+		return bazelWorkspace, nil
+	}
+	return os.Getwd()
```

Comment:
this is unlikely to be the git root? We have the code below in `SuggestGitName` (astore/client/astore/astore.go), basically: git.open current directory, and then get the worktree.

```
        repo, err := git.PlainOpenWithOptions(filepath.Dir(name), &git.PlainOpenOptions{
                DetectDotGit: true,
        })
        if err != nil {
                return "", err
        }

        tree, err := repo.Worktree()
        if err != nil {
                return "", err
        }
        root := tree.Filesystem.Root()

        absRoot, err := filepath.Abs(root)
        if err != nil {
                return "", err
        }

        absName, err := filepath.Abs(name)
        if err != nil {
                return "", err
        }

        destName, err := filepath.Rel(absRoot, absName)
        if err != nil {
                return "", err
        }

        remotes, err := repo.Remotes()
        if err != nil {
                return "", err
        }
```